lib/openstax/rescue_from/wrapped_exception.rb

```diff
@@ -0,0 +1,76 @@
+module OpenStax
+  module RescueFrom
+    class WrappedException
+      attr_reader :exception
+
+      def initialize(exception)
+        @exception = exception
+      end
+
+      def header
+        @header ||= cause.blank? ? 'An exception occurred' : 'Exception cause'
+      end
+
+      def name
+        @name ||= exception.class.name
+      end
+
+      def error_id
+        @error_id ||= Error.id
+      end
+
+      def message
+        @message ||= exception.message
+      end
+
+      def friendly_message
+        config.friendly_status_messages[status]
+      end
+
+      def dns_name
+        @dns_name ||= Resolv.getname(listener.request.remote_ip) rescue 'unknown'
+      end
+
+      def extras
+        @extras ||= if extras_proc = config.exception_extras[exception.class.name]
+                      extras_proc.call(exception)
+                    else
+                      "{}"
+                    end
+      end
+
+      def cause
+        @cause ||= exception.cause
+      end
+
+      def backtrace
+        @backtrace ||= cause.blank? ? all_backtrace_lines : first_backtrace_line
```

Comment:
similar logic reversal?  we want to print all of the backtrace lines for the exception that was rescued, and then the first line of all of the causes, recursively.